src/main/webapp/app/quiz/re-evaluate/quiz-re-evaluate-warning.controller.js

```diff
@@ -61,41 +61,96 @@
-                    question.answerOptions.forEach(function (answer) {
-                        // only check if there are no changes on the answers yet
-                        if (!vm.answerCorrectess || !vm.answerInvalid) {
-                            var backUpAnswer = null;
-                            for (var j = 0; backUpAnswer === null && j < backUpQuestion.answerOptions.length; j++) {
-                                if (backUpQuestion.answerOptions[j].id === answer.id) {
-                                    backUpAnswer = backUpQuestion.answerOptions[j];
+                    //check MultipleChoiceQuestions
+                    if (question.type === "multiple-choice") {
+                        // question-Element deleted?
+                        if (question.answerOptions.length !== backUpQuestion.answerOptions.length) {
+                            vm.questionElementDeleted = true;
+                        }
+
+                        //check each answer
+                        question.answerOptions.forEach(function (answer) {
+                            // only check if there are no changes on the question-elements yet
+                            if (!vm.questionCorrectness || !vm.questionElementInvalid) {
+                                var backUpAnswer = null;
+                                for (var j = 0; backUpAnswer === null && j < backUpQuestion.answerOptions.length; j++) {
+                                    if (backUpQuestion.answerOptions[j].id === answer.id) {
+                                        backUpAnswer = backUpQuestion.answerOptions[j];
+                                    }
+                                }
+                                if (backUpAnswer !== null) {
+                                    //answer set invalid?
+                                    if (answer.invalid !== backUpAnswer.invalid) {
+                                        vm.questionElementInvalid = true;
+                                    }
+                                    //answer correctness changed?
+                                    if (answer.isCorrect !== backUpAnswer.isCorrect) {
+                                        vm.questionCorrectness = true;
+                                    }
+                                }
+                            }
+                        });
+                    }
+                    //check DragAndDropQuestions
+                    if (question.type === "drag-and-drop") {
+                        if (question.dragItems.length !== backUpQuestion.dragItems.length
+                            || question.dropLocations.length !== backUpQuestion.dropLocations.length) {
+                            vm.questionElementDeleted = true;
+                        }
+                        if (!angular.equals(question.correctMappings, backUpQuestion.correctMappings)) {
+                            vm.questionCorrectess = true;
+                        }
+
```

Comment:
please extract the code below into a new method, otherwise it will become too complex